### DISCUSSION TITLE
Sectional snapshots

### DISCUSSION THREAD

**Discussion by glen-84** [VerifyTests/Verify#1180] [Ideas]
Could Verify support sectional snapshots?

For example:

```cs
var snapshot = new Snapshot();
snapshot.AddSection("Input", input, SectionType.Text);
snapshot.AddSection("Output", output, SectionType.Json);

return Verify(snapshot);
```

This is a made-up API, but you get the idea. The snapshot file might look like this:

```
=====
Input
=====

Some text ...

======
Output
======

{
    "some": "json"
}
```

Or, using Markdown for syntax highlighting:

````md
# TestName

## Input

```
Some text ...
```

## Output

```json
{
    "some": "json"
}
```
````

In order to apply ignores, scrubbing, etc., it would need to be supported by Verify directly, since you couldn't take an already-combined file and apply (for example) the JSON processing to it. Each section would have to be processed separately, before being combined into the final snapshot.

**Comment by SimonCropp** [comment]
would this suffice

```
var input = "the input values";
var result = new
{
    Name = "John",
    Age = 21
};
return Verify(
    new
    {
        input,
        result
    });
```

results in 
```
{
  input: the input values,
  result: {
    Name: John,
    Age: 21
  }
}
```

**Reply by glen-84** [reply]
Not really. The idea is for the different parts to be visually separated and to have their own type/format. This makes it easier to read.

With a custom formatter, you could even have syntax highlighting by using Markdown (as previously mentioned).

**Reply by glen-84** [reply]
FWIW, here's an example that uses the Snapshooter/CookieCrumble libraries:

https://github.com/ChilliCream/graphql-platform/blob/0605e2a20e3c1b7d81a0020c1983d6ed0ec0d45b/src/HotChocolate/Fusion/test/Core.Tests/__snapshots__/DemoIntegrationTests.Fetch_User_With_Node_Field.md

**Reply by SimonCropp** [reply]
happy to consider a pull request

**Reply by glen-84** [reply]
Is it possible to run the conversion process on a type within your own file converter?

f.e. If you handle targets of type A, and inside your type A converter, you want to convert A.Prop1 and A.Prop2 as if they were being converted at the top level, running through the process of finding a converter, scrubbing, etc., and returning a string.
